### ISSUE TITLE
"INTERNAL ERROR: QPDF::calculateLinearizationData" with qpdf 8.2.1

### ISSUE THREAD

**Issue by rbrito** [qpdf/qpdf#268]
Hi.

First of all, than you very, very much for qpdf for being such a wonderful manipulation tool of PDF files. It is really appreciated.

OK, now, to the real issue: I have a PDF file that is giving me the following error message when I try to linearize it with qpdf:

    INTERNAL ERROR: QPDF::calculateLinearizationData: page object for page 212 not in lc_other_page_private

Since other tools don't seem to complain with this PDF file and I can process it with qpdf with other functions (like decompressing it), I believe that it may be reasonably well-formed (I believe).

Since the PDF file in question is copyrighted, I am not attaching it here, but I can send it privately.

Thanks,

Rogério Brito.

**Comment by jberkenbilt** [comment]
This is a known issue, and it is on my list of things to do soon, hopefully for the next release or the one right after it. The issue is that the linearization checking code treats a number of malformed linearization constructs as errors instead of warnings. Thanks for your kind words and for taking the time to report this.

**Comment by jberkenbilt** [comment]
@rbrito This is a duplicate of #231, so I am closing this issue. Feel free to subscribe to that one if you'd like updates. Thanks!

**Comment by jberkenbilt** [comment]
@rbrito Feel free to post your files here. I think when I originally read this, I mistook for a duplicate of #231, but it is in fact a separate problem.

**Comment by jberkenbilt** [comment]
And, yes, you may send me files privately. I can delete them when I've solved the issue.

**Comment by jberkenbilt** [comment]
The problem is occurring with a few pages that have zero private objects. These are completely empty pages with empty resource dictionaries and content streams that are shared with other completely empty pages. I'm not entirely sure what the actual problem is though as I can hand-construct simple files that behave in this way, and they linearize as expected. If I simply change the code to not consider this case to be an error, the file linearizes, and the linearized data looks valid at a glance, but I'm not really sure whether it's a correct fix without studying the code and the spec a little more closely. I may spend a little time trying to figure this out in hopes of getting a fix into 8.4.0.

**Comment by jberkenbilt** [comment]
I figured out exactly what's happening: the file has multiple pages that use exactly the same page dictionary. There are other things in the qpdf library that don't work in this case, such as using `--pages` to extract some of the duplicated pages, though in that case, it actually detects it properly. I have now been able to reproduce this with a very small file. I will hang onto the files you sent me privately to verify that whatever fix I come up with handles it. Then I will delete them. Thanks for bringing this to my attention.
